src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriter.cs

```diff
@@ -185,25 +183,31 @@ internal CodeWriter WriteCore(string value, int startIndex, int count)
 
         // Iterate the string, stopping at each occurrence of a newline character. This lets us count the
         // newline occurrences and keep the index of the last one.
-        while ((i = value.IndexOfAny(NewLineCharacters, i)) >= 0)
+        var endIndex = startIndex + count;
+        while (i < endIndex)
         {
-            // Newline found.
-            _currentLineIndex++;
-            _currentLineCharacterIndex = 0;
+            var ch = value[i];
 
             i++;
 
-            // We might have stopped at a \r, so check if it's followed by \n and then advance the index to
-            // start the next search after it.
-            if (count > i &&
-                value[i - 1] == '\r' &&
-                value[i] == '\n')
+            if (ch == '\r' || ch == '\n')
```

Comment:
Bit of a meta question. The VS editor recognizes a few more line break kinds besides `\r` and `\n` like `u2028`. Do we run into any problems here when code uses these alternative line endings?

Reply:
I don't believe it's a problem for us because we're the ones writing the code and the OS drives the line endings. So, if Razor is supported on an OS that isn't CRLF or LF in the future, this would need to be updated, but I don't think that's likely to happen.